contracts/amm/AmmStorage.sol

```diff
@@ -780,6 +795,78 @@ contract AmmStorage is Initializable, PausableUpgradeable, UUPSUpgradeable, Ipor
         );
     }
 
+    function _updateLastOpenedSwapWhenOpenPayFixed(
+        AmmTypes.SwapDuration duration,
+        uint256 swapId,
+        uint256 openTimestamp
+    ) internal {
+        uint32 headSwapId = _openSwapsPayFixed[duration].headSwapId;
+        _openSwapsPayFixed[duration].swaps[swapId.toUint32()] = AmmInternalTypes.OpenSwapItem(
+            swapId.toUint32(),
+            0,
+            headSwapId,
+            openTimestamp.toUint32()
+        );
+        _openSwapsPayFixed[duration].headSwapId = swapId.toUint32();
+        _openSwapsPayFixed[duration].swaps[headSwapId].nextSwapId = swapId.toUint32();
+    }
+
+    function _updateLastOpenedSwapWhenClosePayFixed(AmmTypes.SwapDuration duration, uint256 swapId) internal {
+        uint32 headSwapId = _openSwapsPayFixed[duration].headSwapId;
+        AmmInternalTypes.OpenSwapItem memory swap = _openSwapsPayFixed[duration].swaps[swapId.toUint32()];
+        if (swapId.toUint32() == headSwapId) {
+            AmmInternalTypes.OpenSwapItem memory swapPrev = _openSwapsPayFixed[duration].swaps[swap.previousSwapId];
+            swapPrev.nextSwapId = 0;
+            _openSwapsPayFixed[duration].headSwapId = swapPrev.swapId;
+            _openSwapsPayFixed[duration].swaps[swapPrev.swapId] = swapPrev;
+            delete _openSwapsPayFixed[duration].swaps[swapId.toUint32()];
+        } else {
+            AmmInternalTypes.OpenSwapItem memory swapPrev = _openSwapsPayFixed[duration].swaps[swap.previousSwapId];
```

Comment:
What if swapId is old one, from version 1 of protocol?

Reply:
i added check 
```
if(swap.openSwapTimestamp == 0) {
            return;
        }
```

so we will ignore old one